tests/test_app.py

```diff
@@ -197,118 +232,130 @@ def test_sim_sir():
-    assert chart.data.iloc[1].hospitalized == 1
-    assert chart.data.iloc[49].ventilated == 203
+def test_census_chart():
+    census_df = pd.read_csv("tests/by_doubling_time/2020-03-28_projected_census.csv")
+    chart = build_census_chart(alt=alt, census_df=census_df)
+    assert isinstance(chart, (alt.Chart, alt.LayerChart))
+    assert chart.data.iloc[1].hospitalized == 3
+    assert chart.data.iloc[49].ventilated == 365
 
     # test fx call with no params
     with pytest.raises(TypeError):
-        admitted_patients_chart()
-
-    empty_chart = admitted_patients_chart(alt, pd.DataFrame(), PARAM)
-    assert empty_chart.data.empty
+        build_census_chart()
 
 
 def test_model(model=MODEL, param=PARAM):
     # test the Model
 
-    assert model.infected == 40000.0
+    assert round(model.infected, 0) == 49097.0
     assert isinstance(model.infected, float)  # based off note in models.py
 
     # test the class-calculated attributes
     assert model.detection_probability == 0.125
     assert model.intrinsic_growth_rate == 0.12246204830937302
-    assert model.beta == 3.582761451679408e-07
+    assert model.beta == 4.21501347256401e-07
     assert model.r_t == 2.307298374881539
     assert model.r_naught == 2.7144686763312222
     assert model.doubling_time_t == 7.764405988534983
 
-    # test the things n_days creates, which in turn tests sim_sir, sir, and get_dispositions
-    assert len(model.raw_df) == param.n_days + 1 == 61
 
+def test_model_raw_start(model=MODEL, param=PARAM):
     raw_df = model.raw_df
+
+    # test the things n_days creates, which in turn tests sim_sir, sir, and get_dispositions
+
+    # print('n_days: %s; i_day: %s; n_days_since: %s' % (param.n_days, model.i_day, model.n_days_since))
+    assert len(raw_df) == (param.n_days + model.n_days_since + 1) == 105
+
     first = raw_df.iloc[0, :]
     second = raw_df.iloc[1, :]
-    last = raw_df.iloc[-1, :]
 
     assert first.susceptible + first.infected + first.recovered == param.population
-    assert last.susceptible + last.infected + last.recovered == param.population
-
     assert first.susceptible == 460000.0
     assert round(second.infected, 0) == 43735
-
-    assert round(last.susceptible, 0) == 59497
+    assert list(model.dispositions_df.iloc[0, :]) == [-38, date(year=2020, month=2, day=19), 100.0, 40.0, 20.0]
     assert round(raw_df.recovered[30], 0) == 216711
 
-    assert list(model.dispositions_df.iloc[0, :]) == [0, 100.0, 40.0, 20.0]
-    assert [round(i, 0) for i in model.dispositions_df.iloc[60, :]] == [60, 1101.0, 441.0, 220.0]
+    d, dt, s, i, r = list(model.dispositions_df.iloc[60, :])
+    assert dt == date(year=2020, month=4, day=19)
+    assert [round(v, 0) for v in (d, s, i, r)] == [22, 1101.0, 441.0, 220.0]
+
+
+def test_model_raw_end(model=MODEL, param=PARAM):
+    raw_df = model.raw_df
+
+    last = raw_df.iloc[-1, :]
+    assert last.susceptible + last.infected + last.recovered == param.population
```

Comment:
Test fails here with a difference of 30600 people. Same failure in later raw_begin test.